Documentation (#4)
SABS-R3-Epidemiology/rEpiabm: README.md

ADD BADGES
# rEpiabm

> rEpiabm enables users familiar with R to use [Epiabm](https://github.com/SABS-R3-Epidemiology/epiabm). Epiabm is a simulation tool that models the progress of an epidemic across a specified region of interest within a specific timeframe. It has been developed in python for small-scale implementations and C++ for fast, large-scale simulations. PyEpiabm design is modular, with many options to configure specific requirements.

Reviewer: pyEpiabm has a modular design...

## Summary of Epiabm functionality

### Basic Architecture

> To model an epidemic, contact events are represented by the population spatial structure (see Figure 1.). The transmission of the disease and its progression within host is represented by a compartment model (see Figure 2.). These two architectures are highly configurable; this allows us to study a wide range of simulation scenarios.

Reviewer: ```suggestion
To model an epidemic, contact events occur within the population spatial structure (see Figure 1.). A compartmental model is used for the progression of the disease within-host (see Figure 2.). These two architectures are highly configurable; this allows us to study a wide range of simulation scenarios.
```

<div style="display: flex; gap: 20px;">
  <div style="flex: 1;">
    <img src="./images/population_spatial_structure.png" alt="Population spatial structure" width="100%">
    <p><figcaption><i><b>Figure 1.</b> Population Spatial Structure: The environment is modelled using <a href="https://github.com/SABS-R3-Epidemiology/EpiGeoPop">EpiGeoPop</a>, which takes a region of interest, creates layers of sub-regions of different types and populates these with individuals.</p><figcaption></i>
  </div>
  <div style="flex: 1;">
    <img src="./images/infection_progression.jpg" alt="Infection progression" width="100%">
    <p><figcaption><i><b>Figure 2.</b> Infection Progression: The infection progression is represented using a <a href="https://en.wikipedia.org/wiki/Compartmental_models_in_epidemiology">compartment model</a> which tracks the daily progress of the disease within an individual.</i><figcaption></p>
  </div>
</div>

## Running a simulation

> The basic flow of a simulation is described below, a more detailed, complex example is illustrated in this [jupyter notebook](./walk_through/detailed_example.ipynb). The instructions to run a basic simulation is given and we will use 'Andorra' as the region of interest. Also, [the Wiki](https://github.com/SABS-R3-Epidemiology/epiabm/wiki/Overview-of-the-Ferguson-Model) details optional parameters available to the user as well as those whose values are mentioned, but changing them is not recommended.

Reviewer: ```suggestion
The basic flow of a simulation is described below; a more detailed, complex example is illustrated [in this Jupyter notebook](./walk_through/detailed_example.ipynb). We give instructions to run a basic simulation and use 'Andorra' as the region of interest. Also, [the Wiki](https://github.com/SABS-R3-Epidemiology/epiabm/wiki/Overview-of-the-Ferguson-Model) details optional parameters available to the user as well as those whose values are mentioned, but changing them is not recommended.
```

### Step 1: Set up rEpiabm
Before running a simulation, rEpiabm needs to be installed with all dependencies mentioned in the DESCRIPTION file. Also, the folder structure used by the R program file needs to be set up for your region of interest.

**Instructions:**
1. Go to your github personal Settings/Developer Settings
2. Create a new personal access token (fine-grained)
3. Launch RStudio and type:
```file.edit("~/.Renviron")```

4. In the file add the line:

    ```GITHUB_PAT=your_personal_access_token```

>   where the above is your real token from github.

Reviewer: ```suggestion
  where the above is your real token from Github, and make sure to save the file.
```

5. Restart RStudio

> 6. Enter in R terminal:

Reviewer: ```suggestion
6. Create a new directory for the repository and enter in R terminal (from inside this new directory):
```
